src/ssl.c

```diff
@@ -30956,6 +31431,210 @@ WOLFSSL_ASN1_INTEGER* wolfSSL_BN_to_ASN1_INTEGER(const WOLFSSL_BIGNUM *bn, WOLFS
     return a;
 }
 
+#ifdef OPENSSL_ALL
+void *wolfSSL_ASN1_item_new(const WOLFSSL_ASN1_ITEM *template)
+{
+    void *ret = NULL;
+    const WOLFSSL_ASN1_TEMPLATE *member = NULL;
+    size_t i;
+    WOLFSSL_ENTER("wolfSSL_ASN1_item_new");
+    if (!template) {
+        return NULL;
+    }
+    if (!(ret = XMALLOC(template->size, NULL, DYNAMIC_TYPE_OPENSSL))) {
+        return NULL;
+    }
+    XMEMSET(ret, 0, template->size);
+    for (member = template->members, i = 0; i < template->mcount;
+            member++, i++) {
+        switch(member->type) {
+        case WOLFSSL_X509_ALGOR_ASN1:
+        {
+            WOLFSSL_X509_ALGOR* algor = wolfSSL_X509_ALGOR_new();
+            if (!algor) {
+                goto error;
+            }
+            *(WOLFSSL_X509_ALGOR**)(((byte*)ret) + member->offset) = algor;
```

Comment:
Why is an offset needed?

Reply:
To allow for custom user struct's to be used with `IMPLEMENT_ASN1_FUNCTIONS`.